server/cmd/code-annotation/serializer/serializers.go

```diff
@@ -0,0 +1,112 @@
+package serializer
+
+import (
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/src-d/code-annotation/server/model"
+)
+
+type ResponseErrors struct {
+	Errors []Error `json:"errors"`
+	status int
+}
+
+func (e *ResponseErrors) Render(w http.ResponseWriter, r *http.Request) error {
+	render.Status(r, e.status)
+	return nil
+}
+
+type ResponseSingle struct {
+	Data Identifiable `json:"data"`
+}
+
+func (e *ResponseSingle) Render(w http.ResponseWriter, r *http.Request) error {
+	return nil
+}
+
+type ResponseStored struct {
+	Data int `json:"data"`
+}
+
+func (e *ResponseStored) Render(w http.ResponseWriter, r *http.Request) error {
+	return nil
+}
+
+type ResponseMany struct {
+	Data []Identifiable `json:"data"`
```

Comment:
do you think it's possible we will render something that doesn't have id?
I personally prefer to add constraints only when they are really necessary.
please consider this comment only as an open discussion, not action request.

Reply:
I don't advocate for this particular code, just another way how to do it for you to consider: https://github.com/src-d/code-annotation/pull/22

I also find empty return approach error prone. In my previous project, I myself fixed like 5 bugs about it before we moved to non-empty return. And reviewed about 5 more fixed by other developers. I mean bugs that we found in production, not during CR.

Reply:
I agree with the 1st paragraph.

I'm lost in "_empty return approach_". I could not find any occurrence of `empty` word in this PR. Could you give me a clue?

Reply:
```go
func Handler(w, r) {
  if err != nil {
    write(w)
    return // empty return
  }
  write(w)
}

Reply:
I think what Maxim means is that it's a good and less error-prone to always return something like i.e https://github.com/src-d/code-annotation/pull/22/files#diff-3e10846b2816b527e6724efedecbb964R15

Reply:
As a global assertion, I agree with it... but in the context of `chi` handlers, the expected ones must have the signature [http.HandlerFunc](https://golang.org/pkg/net/http/#HandlerFunc), that is "empty return"
In the code you linked, the empty return handler is deeper, but [it is](https://github.com/src-d/code-annotation/pull/22/files#diff-a2f70ad899ea31772f2946085a294415R39) 

If you're talking about status, errors and so on, I agree, but it's not related to the "empty return" funcs

Reply:
it's not `chi` handlers. It's `net/http` handlers. Of course there is empty return on top level.  But I can't imagine how you can make mistake there. Everything inside won't compile without proper return.

Reply:
`chi.Router.Get(pattern string, h http.HandlerFunc)` requires a `http.HandlerFunc` (an empty return func). It does not matter what you put as `h`, that it'll be an empty return func.

**Again**: I agree on the status, errors and so on, but it's a different thing that I already confirmed it will be added to the server.